x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -512,6 +531,28 @@ public boolean isLogstashAllowed() {
         }
     }
 
+    /**
+     * Beats is allowed as long as there is an active license of type TRIAL, STANDARD, GOLD or PLATINUM
+     * @return {@code true} as long as there is a valid license
+     */
+    public boolean isBeatsAllowed() {
```

Comment:
Given that the ingest team owns, I wonder if this should instead be `isIngestAllowed` and simply replace `isLogstashAllowed`? They share the same license requirements, so it seems reasonable to me that they shouldn't dupe their code too.

Reply:
Personally I feel like the code should reflect the products/features and not organization structure, as much as possible. So I'd like to keep this as-is.

I'd be okay with refactoring the duplicate code in the two methods into a helper method like `isNotBasic` or something like that. What do you think of that?

Reply:
I'd be happy with an `isNotBasic` method.